solver/internal/pipe/pipe.go

```diff
@@ -8,67 +8,67 @@ import (
 	"github.com/pkg/errors"
 )
 
-type channel struct {
+type channel[V any] struct {
 	OnSendCompletion func()
 	value            atomic.Value
```

Comment:
These both can be `atomic.Pointer` I think

Reply:
I think this removed the need for `wrappedValue`. I think `lastValue` doesn't need to be an `atomic.Pointer` and it's just `value`.